kratos/mpi/tests/sources/test_pointer_communicator.cpp

```diff
@@ -186,9 +187,67 @@ KRATOS_TEST_CASE_IN_SUITE(PointerCommunicatorConstructByFunctor, KratosMPICoreFa
     }
 
 
-};
+}
 
+KRATOS_TEST_CASE_IN_SUITE(PointerMapCommunicator, KratosMPICoreFastSuite)
+{
+    DataCommunicator& r_default_comm = ParallelEnvironment::GetDefaultDataCommunicator();
+    Model current_model;
+    auto& mp = current_model.CreateModelPart("mp");
+    mp.AddNodalSolutionStepVariable(PARTITION_INDEX);
+
+    const int world_size = r_default_comm.Size();
+    const int current_rank = r_default_comm.Rank();
 
+    auto pnode = mp.CreateNewNode(current_rank+1, current_rank,current_rank,current_rank); //the node is equal to the current rank;
+    pnode->FastGetSolutionStepValue(PARTITION_INDEX) = current_rank;
+    pnode->SetValue(TEMPERATURE, current_rank);
+
+    //we will gather on every node the global pointers of the nodes with index from
+    //current_rank(+1) to world_size
+    std::vector<int> indices;
+    for(int i=1; i<=world_size; ++i) {
+        indices.push_back(i);
+    }
+
+    auto gp_list = GlobalPointerUtilities::RetrieveGlobalIndexedPointers(mp.Nodes(), indices, r_default_comm);
+
+    // creates the gp map
+    GlobalPointerMapCommunicator<Node<3>, double> pointer_map_comm(r_default_comm, gp_list, 0.0);
+
+    // creates apply proxy
+    auto apply_temperature_assemble_proxy = pointer_map_comm.GetApplyProxy(
+        // local gp value update method
+        [](Node<3>& rNode, const double& NewValue) {
+            rNode.GetValue(TEMPERATURE) += NewValue;
+        },
+        // non local gp map update lambda method
+        [](double& rCurrentValue, const double NewValue) {
+            rCurrentValue += NewValue;
+        });
+
+    GlobalPointersUnorderedMap<Node<3>, double> assembly_values_map;
```

Comment:
with the proposed implementation you need to store both gp_list and gp_map.
Can't u do only with gp_map?

Reply:
This can be used without a gp_map, I put gp_map as an option only. This can be directly used with GPs as in the following and in the case of serial, it will be least cost since it directly calls the update. 
https://github.com/KratosMultiphysics/Kratos/blob/ddefe50961014ff288afb7099c213cbfb02af399/kratos/mpi/tests/sources/test_pointer_communicator.cpp#L237